docs/docker.rst

```diff
@@ -0,0 +1,160 @@
+Docker is an application that makes it simple and easy to run processes in a container,
+which are like virtual machines, more resource-friendly. For a detailed introduction to the
+different components of a Docker container.
+
+Useful `Tutorial
+<https://www.digitalocean.com/community/tutorials/the-docker-ecosystem-an-introduction-to-common-components>`_.
+
+
+Inspire and Docker
+##################
+
+Get the latest Docker setup.
+
+.. note:: If you are using Mac, please build a simple box with ``docker-engine`` above ``1.10`` and
+         ``docker-compose`` above ``1.6.0``.
+
+Make sure you can run docker without ``sudo``.
+
+- ``id $USER``
+
+  If you are not in the ``docker`` group, run the following command and then restart ``docker``. If this doesn't work, just restart your machine :)
+
+- ``newgrp docker`` or ``su - $USER``
+
+- ``sudo usermod -a -G docker $USER``
+
+Get `docker-compose
+<https://docs.docker.com/compose/>`_:
+
+.. code-block:: console
+
+ sudo pip install docker-compose
+
+- Add ``DOCKER_DATA`` env variable in your ``.bashrc`` or ``.zshrc``. In this directory you will have all the persistent data between Docker runs.
+
+.. code-block:: console
+
+ shell mkdir -p "$DOCKER_DATA"
+ export DOCKER_DATA=~/inspirehep_docker_data/
+
+By default the virtualenv and everything else will be kept on ``/tmp`` and they will be available only until the next reboot.
+
+- Install a host persistent venv and build assets
+
+.. code-block:: console
+
+ docker-compose pull
+ docker-compose -f docker-compose.deps.yml run --rm pip
+
+.. note:: If you have trouble with internet connection inside docker probably you are facing known
+          DNS issue. Please follow `this solution
+          <http://askubuntu.com/questions/475764/docker-io-dns-doesnt-work-its-trying-to-use-8-8-8-8/790778#790778>`_
+          with DNS: ``--dns 137.138.17.5 --dns 137.138.16.5``.
+
+.. code-block:: console
+
+ docker-compose -f docker-compose.deps.yml run --rm assets
+
+- Run the service locally
+
+.. code-block:: console
+
+ docker-compose up
+
+Go to ``localhost:5000``
+
+- Populate database
+
+.. code-block:: console
+
+ docker-compose run --rm web scripts/recreate_records
+
+- Run tests in an **isolated** environment.
+
+  You can choose one of the following tests types:
+
+  - unit
+  - disambiguation
+  - workflows
+  - integration
+  - acceptance
+
+.. code-block:: console
+
+ docker-compose -f docker-compose.test.yml run --rm <tests type>
+ docker-compose -f docker-compose.test.yml down
+
+.. tip:: - cleanup all the containers:
+
+           ``docker rm $(docker ps -qa)``
+
+         - cleanup all the images:
+
+           ``docker rmi $(docker images -q)``
+
+         - cleanup the virtualenv (careful, if docker_data is set to something you care about, it will be removed):
+
+           ``sudo rm -rf "${DOCKER_DATA?DOCKER_DATA was not set, ignoring}"``
+
+Extra useful tips
+#################
+
+- Run a random shell
+
+After ``docker-compose up`` just run:
```

Comment:
Yeah it's not need. The only reason that we want it its just for monitoring/logging purposes.